db/consistency_level.cc

```diff
@@ -352,15 +352,6 @@ filter_for_query(consistency_level cl,
     return selected_endpoints;
 }
 
-inet_address_vector_replica_set filter_for_query(consistency_level cl,
```

Comment:
I do not see the benefit of this patch. Now we have `filter_for_query` in `consistency_level` and in the `storage_proxy`.

Reply:
The goal is to centralize accesses to `_gossiper` in `storage_proxy`, to eventually put them behind `remote`.

Other than that, we reduce the number of undocumented public APIs - this `filter_for_query` overload in db/consistency_level.hh, which really existed only for the single purpose of making code in `storage_proxy` a bit shorter (passing fewer args). Since that was its purpose, I think it belongs in `storage_proxy` better.

Reply:
> Why do you need to move the function for that?

If you have an alternative, please provide.

> The argument that it is used only once is not enough to justify spreading consistency_level API over several files.

This is not what I'm doing, `storage_proxy::filter_for_query` is private. If there are other users of `filter_for_query` outside `storage_proxy` they see a simple API consisting of a single function in db/consistency_level.hh.

Reply:
> > Why do you need to move the function for that?
> 
> If you have an alternative, please provide.

The alternative of moving something is not moving it.

> 
> > The argument that it is used only once is not enough to justify spreading consistency_level API over several files.
> 
> This is not what I'm doing, `storage_proxy::filter_for_query` is private. If there are other users of `filter_for_query` outside `storage_proxy` they see a simple API consisting of a single function in db/consistency_level.hh.

Private has nothing to so with it. I prefer to see everything related to consistency in one place.  I am sure we can merge more than half of `db/consistency_level` into the storage proxy because, by it designation, it is the main user of the API, but I do not think it should be done just because we can.

Reply:
> The alternative of moving something is not moving it.

Yes, very productive. Thank you.

I said:
> The goal is to centralize accesses to _gossiper in storage_proxy, to eventually put them behind remote.

You asked:
> Why do you need to move the function for that?

So I asked for an alternative to solve the problem I was solving. I wasn't moving stuff just for the heck of it, I was moving stuff to solve a problem: "centralize accesses to _gossiper in storage_proxy", as I explained.

So I'm gonna ask again, since apparently my previous question wasn't very obvious:
If you have an alternative solution in order to **centralize accesses to `_gossiper` in `storage_proxy`**, other than moving code around, please provide the solution.

Reply:
Are you proposing to remove the part of the patch which removes the `db::filter_for_query` overload, but leave the part of the patch which adds `storage_proxy::filter_for_query`? Then instead of "moving" a function we will "add" a function.

Reply:
@gleb-cloudius reading the comments, if you have a specific suggestion, like keeping the overload in db/consistency_level.hh, and, then adding a matching overload to storage_proxy.cc, please make it.

Reply:
> Are you proposing to remove the part of the patch which removes the `db::filter_for_query` overload, but leave the part of the patch which adds `storage_proxy::filter_for_query`? Then instead of "moving" a function we will "add" a function.

May be I do not understand the problem you are trying to solve, but then commit message does not explain it at all. The commit message says the goal of the patch is: "to reduce the number of different functions that directly access `_gossiper` in `storage_proxy`". If this is the only reason then my solution is to scrap the patch. Call the function from where it is now directly.